Packs/Workday/Integrations/WorkdayIAM/WorkdayIAM.py

```diff
@@ -0,0 +1,294 @@
+        self._report_url = report_url
+
+    # Getting Workday Full User Report with a given report URL. This uses RaaS
+    def get_full_report(self):
+        return self._http_request(method="GET", full_url=self._report_url, url_suffix="",
+                                  timeout=READ_TIME_OUT_IN_SECONDS)
+
+
+def convert_incident_fields_to_cli_names(data):
+    converted_data = {}
+    for k, v in data.items():
+        key_machine_name = k.lower().replace(' ', '')
+        converted_data[key_machine_name] = v
+    return converted_data
+
+
+def get_time_elapsed(fetch_time, last_run):
+    now = datetime.now()
+    demisto.info("Workday Last Run: " + str(last_run))
+    if 'time' in last_run:
+        # Get Last run and parse to date format. Workday report will be pulled from last run time to current time
+        last_run_time = last_run['time']
+        # Convert to date format
+        last_run = datetime.strptime(last_run_time, WORKDAY_DATE_TIME_FORMAT)
+        time_elapsed_in_minutes = (now - last_run).total_seconds() / 60
+    else:
+        # If last run time is not set, data will be pulled using fetch_time
+        # i.e. last 10min if fetch events is set to 10min
+        last_run_time = (now - timedelta(minutes=int(fetch_time))).strftime(
+            WORKDAY_DATE_TIME_FORMAT)
+        time_elapsed_in_minutes = fetch_time
+
+    return time_elapsed_in_minutes, last_run_time
+
+
+def fetch_incidents(client, last_run, fetch_time, mapper_in):
+    """
+    This function will execute each interval (default is 1 minute).
+
+    Args:
+        client: Workday client
+        last_run: The greatest incident created_time we fetched from last fetch
+        fetch_time: The time interval when the function should execute and return events/incidents
+
+    Returns:
+        last_run: This will be last_run in the next fetch-incidents
+        events: Incidents/Events that will be created in Cortex XSOAR
+    """
+    start = datetime.now()
+    events = []
+    from_date_time = '###'
+    to_date_time = '$$$'
+    try:
+        # If there is no fetch time configured, it will be set to 0 and no events will be pulled
+        fetch_time = int(fetch_time) if fetch_time else 0
+        time_elapsed_in_minutes, last_run_time = get_time_elapsed(fetch_time, last_run)
+
+        from_date_time = last_run_time
+        if fetch_time != 0 and time_elapsed_in_minutes >= fetch_time:
+            to_date_time = datetime.now().strftime(WORKDAY_DATE_TIME_FORMAT)
+            report_data = client.get_full_report()
+            report_entries = report_data.get('Report_Entry')
+            for entry in report_entries:
+                workday_user = demisto.mapObject(entry, mapper_in, INCIDENT_TYPE)
+                workday_user = convert_incident_fields_to_cli_names(workday_user)
+                demisto_user = get_demisto_user(workday_user)
```

Comment:
@dantavori @ohaim1008 I suggest to query for all userprofiles in xsoar, and then compare in this list. Otherwise we will do a lot of queries to indicators.
`type:\"User Profile\"` - create a function that query all the users (with paging of 1000 each time in it) and return all the users in dict. The key will be the email/employee id and the value the whole UserProfile